rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java

```diff
@@ -973,6 +971,190 @@ void saveSvpSpendTxWaitingForSignatures_postLovell700AndNullSvpSpendTxWaitingFor
+                SVP_SPEND_TX_WAITING_FOR_SIGNATURES.getKey(),
+                BridgeSerializationUtils.serializeRskTxWaitingForSignatures(svpSpendTxWaitingForSignatures));
+
+            // Act
+            Optional<Map.Entry<Keccak256, BtcTransaction>> actualSvpSpendTxWaitingForSignatures =
+                bridgeStorageProvider.getSvpSpendTxWaitingForSignatures();
+
+            // Assert
+            assertEquals(Optional.empty(), actualSvpSpendTxWaitingForSignatures);
+        }
+
+        @Test
+        void getSvpSpendTxWaitingForSignatures_whenThereIsNoSvpSpendTxWaitingForSignaturesSaved_shouldReturnEmpty() {
+            Optional<Map.Entry<Keccak256, BtcTransaction>> actualSvpSpendTxWaitingForSignatures =
+                bridgeStorageProvider.getSvpSpendTxWaitingForSignatures();
+            assertEquals(Optional.empty(), actualSvpSpendTxWaitingForSignatures);
+        }
+
+        @Test
+        void getSvpSpendTxWaitingForSignatures_whenEntrySetButNotSavedToStorage_shouldReturnTheHash() {
+            // Arrange
+            bridgeStorageProvider.setSvpSpendTxWaitingForSignatures(svpSpendTxWaitingForSignatures);
+
+            // Act
+            Optional<Map.Entry<Keccak256, BtcTransaction>> actualSvpSpendTxWaitingForSignatures =
+                bridgeStorageProvider.getSvpSpendTxWaitingForSignatures();
+
+            // Assert
+            assertTrue(actualSvpSpendTxWaitingForSignatures.isPresent());
+            assertEquals(svpSpendTxWaitingForSignatures, actualSvpSpendTxWaitingForSignatures.get());
+        }
+
+        @Test
+        void getSvpSpendTxWaitingForSignatures_whenDifferentEntryIsInStorageAndAnotherIsSetButNotSaved_shouldReturnTheSetEntry() {
+            // Arrange
+            Keccak256 anotherSpendTxCreationHash = RskTestUtils.createHash(2);
+            BtcTransaction anotherSvpSpendTx = new BtcTransaction(mainnetBtcParams);
+            Map.Entry<Keccak256, BtcTransaction> anotherSvpSpendTxWaitingForSignatures =
+              new AbstractMap.SimpleEntry<>(anotherSpendTxCreationHash, anotherSvpSpendTx);
+            repository.addStorageBytes(
+                bridgeAddress,
+                SVP_SPEND_TX_WAITING_FOR_SIGNATURES.getKey(),
+                BridgeSerializationUtils.serializeRskTxWaitingForSignatures(anotherSvpSpendTxWaitingForSignatures));
+            bridgeStorageProvider.setSvpSpendTxWaitingForSignatures(svpSpendTxWaitingForSignatures);
+
+            // Act
+            Optional<Map.Entry<Keccak256, BtcTransaction>> actualSvpSpendTxWaitingForSignatures =
+                bridgeStorageProvider.getSvpSpendTxWaitingForSignatures();
+
+            // Assert
+            assertTrue(actualSvpSpendTxWaitingForSignatures.isPresent());
+            assertEquals(svpSpendTxWaitingForSignatures, actualSvpSpendTxWaitingForSignatures.get());
+        }
+
+        @Test
+        void getSvpSpendTxWaitingForSignatures_whenStorageIsNotEmptyAndEntrySetToNullButNotSaved_shouldReturnEmpty() {
```

Comment:
nit: since this test is testing the same behavior as the one above (but with null), i would have both names with the same structure